### ISSUE TITLE
Auto-generated mutation payload incorrectly set to null when mutation returns null

### ISSUE THREAD

**Issue by nancyl-ol** [ChilliCream/graphql-platform#5476]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

We use `.AddMutationConventions(applyToAllMutations: true)` to automatically apply conventions to all our mutations. However, when `null` is returned from the mutation, it seems to return `null` for the entire auto-generated `__payload` type.

For example:

```
// mutation code
public async Task<GuardianProfile> DeleteProfileAndRelationships

// generated schema
deleteProfileAndRelationships: DeleteProfileAndRelationshipsPayload!

// error when DeleteProfileAndRelationships returns null GuardianProfile
{
  "errors": [
    {
      "message": "Cannot return null for non-nullable field.",
      "locations": [
        {
          "line": 4,
          "column": 3
        }
      ],
      "path": [
        "deleteProfileAndRelationships"
      ],
      "extensions": {
        "code": "HC0018"
      }
    }
  ]
}
```

**Expected**:
Valid `DeleteProfileAndRelationshipsPayload` returned with a `guardianProfile: null` field

**Actual**: 
Nullability error thrown

### Steps to reproduce

1. Use `.AddMutationConventions(applyToAllMutations: true)`
2. Create a simple mutation that returns a reference type
3. Observe that a non-nullable `___payload` type is auto-generated in the schema
4. Return null from mutation
5. Observe that a nullability error is thrown
6. Now, use `[UseMutationConvention(Disable = true)]` to NOT use mutation convention
7. Observe that a nullable return type is specified in the schema, as expected
8. Return null from mutation
9. Observe that it works as expected, null is returned

### Product

Hot Chocolate

### Version

12.7.0

**Comment by michaelstaib** [comment]
This is not a bug. It's GraphQL spec behavior. Watch https://youtu.be/Zx0nvTUfjn4

**Comment by michaelstaib** [comment]
But why does it work without convention. It should also fail

**Comment by nancyl-ol** [comment]
> But why does it work without convention. It should also fail

`GuardianProfile` is a nullable type since it's a reference type and we don't have nullable reference types enabled (https://chillicream.com/docs/hotchocolate/v12/defining-a-schema/non-null#implicit-nullability). So without the convention, the mutation is allowed to return null.

With the convention enabled, the auto-generated mutation return type `DeleteProfileAndRelationshipsPayload` is non-nullable, which makes sense. However, when the mutation returns a null` GuardianProfile`, I am expecting
```
{
  "data": {
    "deleteProfileAndRelationships": {
      "guardianProfile": null
    }
  }
}
```

Instead, I think it is returning null for the entire `DeleteProfileAndRelationships**Payload**`, which causes the nullability error.


> Can you create a minimal Repro?

Sure, working on creating one

**Comment by nancyl-ol** [comment]
@michaelstaib minimum repro here: https://github.com/nancyl-ol/hotchocolate-5476-repro/tree/main/Demo

Note that nullable reference types is NOT enabled for the repro.

**Comment by michaelstaib** [comment]
@nancyl-ol  tanks for the repro. The issue here seems to be that this mutation has now `input`. Its a bug ... By trying to repro your issue I found another related bug. I will fix both with 12.16.0

I will add some test cases first and you can expect a fix by the end of the week.

**Comment by michaelstaib** [comment]
It turns out that we have fixed the issue already with version 13 preview 68. I will do a backport to 12 for that since we 13 is still 2 month out.
